src/blocks/scratch3_looks.js

```diff
@@ -343,6 +343,8 @@ class Scratch3LooksBlocks {
             } else if (requestedCostume === 'next costume' ||
                        requestedCostume === 'next backdrop') {
                 target.setCostume(target.currentCostume + 1);
+            } else if (requestedCostume === 'random backdrop') {
+                target.setCostume(Math.floor(Math.random() * target.getCostumes().length));
```

Comment:
I think it would be nice to skip the currently selected backdrop when choosing a random one to switch to. That way, every time the user executes the switch backdrop to random block, it always switches backdrops (if there's more than one available).

You can do this by calculating a random number between 0 and one less than the number of backdrops, and then, if the selected random number is greater than or equal to the currently selected costume index (`target.currentCostume`), add 1 to it. The setCostume function will already handle an input that's out of bounds.

Reply:
> Then, if the selected random number is greater than or equal to the currently selected costume index (target.currentCostume), add 1 to it.

Would this cause issues if the current costume is the last costume? So adding 1 to the costume number would make it switch to a nonexistant costume. You could make it so that it `%`'s the picked index plus one with the number of costumes, so that in this case it would select `1` instead of `(number of costumes)`. But then it would pick `1` more often than the other numbers, because there would be two cases that could result in `1`: randomly selecting `1`, or randomly selecting the current costume index which is the last costume.

A solution would be to keep picking random indexes while the picked index is equal to the current index:

```js
let index;
do {
  index = Math.floor(Math.random() * target.getCostumes().length);
} while (index === target.currentCostume);
target.setCostume(index);
```

You would also have to make sure that there *is* more than one costume to select, though; otherwise it would be a never-ending loop!

Reply:
@towerofnix, sorry for the late response, but the proposed solution does not cause problems if the currentCostume is the last costume. We'd be calculating a random number capping at one less than the last index, so that adding 1 to it only gives us the last index. `setCostume` clamps inputs that are out of bounds anyways. see above for the `next backdrop` and `previous backdrop` cases which already rely on the `setCostume` clamp working.

Reply:
Ah, got it! Sounds good.